astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,103 @@
+import astropy.cosmology.units as cu
+import astropy.units as u
+from astropy.cosmology import Parameter, Planck18
+from astropy.cosmology.connect import readwrite_registry
+from astropy.cosmology.core import Cosmology
+from astropy.table import QTable
+
+from .table import from_table, to_table
+
+
+def read_table(filename, index=None, *, move_to_meta=False, cosmology=None, **kwargs):
+    """Read a `~astropy.cosmology.Cosmology` from an HTML file.
+
+    Parameters
+    ----------
+    filename : path-like or file-like
+        From where to read the Cosmology.
+    index : int, str, or None, optional
+        Needed to select the row in tables with multiple rows. ``index`` can be
+        an integer for the row number or, if the table is indexed by a column,
+        the value of that column. If the table is not indexed and ``index``
+        is a string, the "name" column is used as the indexing column.
+
+    move_to_meta : bool (optional, keyword-only)
+        Whether to move keyword arguments that are not in the Cosmology class'
+        signature to the Cosmology's metadata. This will only be applied if the
+        Cosmology does NOT have a keyword-only argument (e.g. ``**kwargs``).
+        Arguments moved to the metadata will be merged with existing metadata,
+        preferring specified metadata in the case of a merge conflict
+        (e.g. for ``Cosmology(meta={'key':10}, key=42)``, the ``Cosmology.meta``
+        will be ``{'key': 10}``).
+
+    cosmology : str, `~astropy.cosmology.Cosmology` class, or None (optional, keyword-only)
+        The cosmology class (or string name thereof) to use when constructing
+        the cosmology instance. The class also provides default parameter values,
+        filling in any non-mandatory arguments missing in 'table'.
+
+    **kwargs
+        Passed to :attr:`astropy.table.QTable.read`
+
+    Returns
+    -------
+    `~astropy.cosmology.Cosmology` subclass instance
+    """
+    kwargs["format"] = "ascii.html"
+    with u.add_enabled_units(cu):
+        table = QTable.read(filename, **kwargs)
+
+    # build cosmology from table
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+def write_table(cosmology, file, *, overwrite=False, cls=QTable, format_dict=None, cosmology_in_meta=True, **kwargs):

```

Comment:
Thanks for the feedback! This was my first time using Flake8, so it might take a little getting used to. I did run the command for Flake8, but it looks there may be other formatting issues, especially with the test. I will try to fix those in the next update.

Reply:
If you want a tool that will take care of most of the flake8 issues, try

```
conda install autopep8
autopep8 --in-place --aggressive --aggressive <filename>
```

Reply:
Be sure to only specify the 1 file you want to modify!

Reply:
Autopep8 really helped to pass those tox checks. Thanks!